### PR TITLE
feat(stepper): add stepper

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**

There is no common _Stepper_ component.

**What is the chosen solution to this problem?**

Extract and adapt the one used in _ui-dataset_.

**Please check if the PR fulfills these requirements**

* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: output/stepper.eslint.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+   81:21  error  A function with a name starting with an uppercase letter should only be used as a constructor  new-cap
+  132:21  error  A function with a name starting with an uppercase letter should only be used as a constructor  new-cap
+  146:17  error  A function with a name starting with an uppercase letter should only be used as a constructor  new-cap
+  158:25  error  A function with a name starting with an uppercase letter should only be used as a constructor  new-cap
+  216:25  error  A function with a name starting with an uppercase letter should only be used as a constructor  new-cap
+  276:31  error  A function with a name starting with an uppercase letter should only be used as a constructor  new-cap
+  277:25  error  A function with a name starting with an uppercase letter should only be used as a constructor  new-cap
+  332:25  error  A function with a name starting with an uppercase letter should only be used as a constructor  new-cap
+
+✖ 14 problems (14 errors, 0 warnings)
```
Comment (smouillour) [inline output/stepper.eslint.txt]: maybe not good to start a new package with 14 errors?

File: packages/stepper/src/components/Stepper.connect.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { connect } from 'react-redux';
+import { default as Stepper } from './Stepper.component';
+import { getStepsForResource } from '../service/Stepper.selectors';
+
+export const mapStateToProps = (state, ownProps) => ({
+	steps: getStepsForResource(state, ownProps.resourceType, ownProps.resourceId),
+});
+
+export default connect(mapStateToProps)(Stepper);
```
Comment (smouillour) [inline packages/stepper/src/components/Stepper.connect.js]: this is ok to have a container in a components folder? 
i know that will be strange to have a folder containers with only this file but in this case is also strange for me :s

Comment (frassinier) [reply packages/stepper/src/components/Stepper.connect.js]: I will move into a new folder :+1:

File: output/stepper.eslint.txt (hunk 1 above)
Comment (frassinier) [reply output/stepper.eslint.txt]: I will fix them indeed

**Comment by jsomsanith-tlnd** [comment]
If I'm not wrong, we use yarn workspaces with lerna, which means the package must be added in package.json > workspaces

File: packages/stepper/package.json [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    "@babel/plugin-proposal-object-rest-spread": "^7.2.0",
+    "@babel/polyfill": "^7.0.0",
+    "@babel/preset-env": "^7.2.0",
+    "@babel/preset-react": "^7.0.0",
+    "@storybook/addon-a11y": "^5.1.0",
+    "@storybook/addon-actions": "^5.1.0",
+    "@storybook/addons": "^5.1.0",
+    "@storybook/react": "^5.1.0",
+    "@talend/icons": "^4.4.0",
+    "@talend/react-components": "^4.4.0",
```
Comment (jsomsanith-tlnd) [inline packages/stepper/package.json]: Can you put all `@talend` packages as dep (not dev-dep) please ?

File: packages/stepper/src/service/Stepper.utils.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ */
+export const isErrorInSteps = steps =>
+	!!steps.find(step => step.status === LOADING_STEP_STATUSES.FAILURE);
+
+/**
+ * This function tells if all the steps are successful
+ * @param {array} steps array of steps
+ */
+export const isAllSuccessful = steps =>
+	steps.filter(step => step.status === LOADING_STEP_STATUSES.SUCCESS).length === steps.length;
```
Comment (jsomsanith-tlnd) [inline packages/stepper/src/service/Stepper.utils.js]:
```suggestion
	steps.every(step => step.status === LOADING_STEP_STATUSES.SUCCESS);
```

File: packages/stepper/src/service/Stepper.utils.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ * @param {object} action redux action
+ */
+export const getStepperKey = action => `${action.resourceType}-${action.resourceId}`;
+
+/**
+ * This function tells if there is an error in the steps
+ * @param {array} steps array of steps
+ */
+export const isErrorInSteps = steps =>
+	!!steps.find(step => step.status === LOADING_STEP_STATUSES.FAILURE);
```
Comment (jsomsanith-tlnd) [inline packages/stepper/src/service/Stepper.utils.js]:
```suggestion
	steps.some(step => step.status === LOADING_STEP_STATUSES.FAILURE);
```

File: packages/stepper/package.json (hunk 3 above)
Comment (frassinier) [reply packages/stepper/package.json]: No because it's managed by the final app, isn't it?

**Pull request merged**